### PR TITLE
override hostname in mesos

### PR DESCRIPTION
**Pull request by vieux**
so it should work with our from scratch image

### CODE REVIEW

**Comment by vieux** [comment]
ping @jimenez

File: cluster/mesos/cluster.go [hunk 1]
```diff
@@ -72,10 +73,15 @@ func NewCluster(scheduler *scheduler.Scheduler, store *state.Store, TLSConfig *t
 	// Empty string is accepted by the scheduler.
 	user, _ := options.String("mesos.user", "SWARM_MESOS_USER")
 
+	hostname, _ := os.Hostname()
+	if hostname == "" {
+		hostname = frameworkName
```
Comment (tnachen) [inline cluster/mesos/cluster.go]: why framework name?

Comment (vieux) [inline cluster/mesos/cluster.go]: @tnachen because I needed something :D do you have another idea ?

Comment (tnachen) [inline cluster/mesos/cluster.go]: hmm, I think if we can't get the hostname and there are no overrides, I wonder if we can just don't pass override?
Passing framework name doesn't even seem logical to me..

Comment (vieux) [inline cluster/mesos/cluster.go]: the official image we produce has no hostname. we need to give something otherwise mesos-go fails

**Comment by vieux** [comment]
@tnachen PTAL

**Comment by jimenez** [comment]
LGTM

**Pull request merged**